--- a/main/java/net/minecraft/client/gui/screen/ConnectingScreen.java
+++ b/main/java/net/minecraft/client/gui/screen/ConnectingScreen.java
@@ -21,6 +21,7 @@
 import net.minecraft.util.text.TranslationTextComponent;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import si.f5.pa_union.reconnect.Reconnect;
 
 public class ConnectingScreen extends Screen
 {
@@ -41,6 +42,9 @@
         mcIn.unloadWorld();
         mcIn.setServerData(serverDataIn);
         this.connect(serveraddress.getIP(), serveraddress.getPort());
+        Reconnect.serverIP = serverDataIn.serverIP;
+        Reconnect.lastScreen = parent;
+        Reconnect.isLan = serverDataIn.isOnLAN();
     }
 
     public ConnectingScreen(Screen parent, Minecraft mcIn, String hostName, int port)
@@ -50,6 +54,9 @@
         this.previousGuiScreen = parent;
         mcIn.unloadWorld();
         this.connect(hostName, port);
+        Reconnect.serverIP = hostName + ":" + port;
+        Reconnect.lastScreen = parent;
+        Reconnect.isLan = false;
     }
 
     private void connect(final String ip, final int port)
