--- a/main/java/net/minecraft/client/Minecraft.java
+++ b/main/java/net/minecraft/client/Minecraft.java
@@ -237,6 +237,10 @@
 import net.minecraft.world.storage.ServerWorldInfo;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import si.f5.pa_union.KurarClient;
+import si.f5.pa_union.discord.DiscordRPCLink;
+import zsawyer.mods.mumblelink.MumbleLinkImpl;
+import zsawyer.mods.mumblelink.UpdateTicker;
 
 public class Minecraft extends RecursiveEventLoop<Runnable> implements ISnooperInfo, IWindowEventListener
 {
@@ -394,6 +398,7 @@
         this.field_244734_au = this.func_244735_a(yggdrasilauthenticationservice, gameConfig);
         this.session = gameConfig.userInfo.session;
         LOGGER.info("Setting user: {}", (Object)this.session.getUsername());
+        DiscordRPCLink.UserName = this.session.getUsername();
         LOGGER.debug("(Session ID is {})", (Object)this.session.getSessionID());
         this.isDemo = gameConfig.gameInfo.isDemo;
         this.enableMultiplayer = !gameConfig.gameInfo.disableMultiplayer;
@@ -458,6 +463,7 @@
         this.framebuffer = new Framebuffer(this.mainWindow.getFramebufferWidth(), this.mainWindow.getFramebufferHeight(), true, IS_RUNNING_ON_MAC);
         this.framebuffer.setFramebufferColor(0.0F, 0.0F, 0.0F, 0.0F);
         this.resourceManager = new SimpleReloadableResourceManager(ResourcePackType.CLIENT_RESOURCES);
+        new KurarClient();
         this.resourcePackRepository.reloadPacksFromFinders();
         this.gameSettings.fillResourcePackList(this.resourcePackRepository);
         this.languageManager = new LanguageManager(this.gameSettings.language);
@@ -550,13 +556,8 @@
 
     private String getWindowTitle()
     {
-        StringBuilder stringbuilder = new StringBuilder("Minecraft");
+        StringBuilder stringbuilder = new StringBuilder("Kurar Client");
 
-        if (this.isModdedClient())
-        {
-            stringbuilder.append("*");
-        }
-
         stringbuilder.append(" ");
         stringbuilder.append(SharedConstants.getVersion().getName());
         ClientPlayNetHandler clientplaynethandler = this.getConnection();
@@ -568,24 +569,32 @@
             if (this.integratedServer != null && !this.integratedServer.getPublic())
             {
                 stringbuilder.append(I18n.format("title.singleplayer"));
+                DiscordRPCLink.details = I18n.format("title.singleplayer");
             }
             else if (this.isConnectedToRealms())
             {
                 stringbuilder.append(I18n.format("title.multiplayer.realms"));
+                DiscordRPCLink.details = I18n.format("title.multiplayer.realms");
             }
             else if (this.integratedServer == null && (this.currentServerData == null || !this.currentServerData.isOnLAN()))
             {
                 stringbuilder.append(I18n.format("title.multiplayer.other"));
+                DiscordRPCLink.details = I18n.format("title.multiplayer.other");
             }
             else
             {
                 stringbuilder.append(I18n.format("title.multiplayer.lan"));
+                DiscordRPCLink.details = I18n.format("title.multiplayer.lan");
             }
+        } else {
+            DiscordRPCLink.details = "In Menu";
         }
 
+        DiscordRPCLink.update();
         return stringbuilder.toString();
     }
 
+
     private SocialInteractionsService func_244735_a(YggdrasilAuthenticationService p_244735_1_, GameConfiguration p_244735_2_)
     {
         try
@@ -1875,7 +1884,11 @@
         this.profiler.endStartSection("keyboard");
         this.keyboardListener.tick();
         this.profiler.endSection();
+
+        if(UpdateTicker.i) {
+            MumbleLinkImpl.instance.getUpdateTicker().tickEnd();
     }
+    }
 
     private boolean func_244600_aM()
     {
